app/app.go

```diff
@@ -42,6 +42,9 @@ type App interface {
 
 	ProvisionDevice(ctx context.Context, dev model.NewDevice) error
 	DecommissionDevice(ctx context.Context, devID uuid.UUID) error
+
+	ConfigurationSet(ctx context.Context, devID uuid.UUID, configuration model.Configuration) error
```

Comment:
Again, `SetConfiguration` is a better name, IMHO.

Reply:
changed.